restapi/src/test/java/io/stargate/web/docsapi/service/query/ExpressionParserIntTest.java

```diff
@@ -131,6 +136,124 @@ public void andWrappingRespected() {
                                     }));
               });
     }
+
+    @Test
+    public void complex() {
+      // a & (b | (c & d))
+      ObjectNode root = mapper.createObjectNode();
+      root.set("a", mapper.createObjectNode().put("$eq", "a"));
+      root.set(
+          "$or",
+          mapper
+              .createArrayNode()
+              .add(mapper.createObjectNode().set("b", mapper.createObjectNode().put("$eq", "b")))
+              .add(
+                  mapper
+                      .createObjectNode()
+                      .set(
+                          "$and",
+                          mapper
+                              .createArrayNode()
+                              .add(
+                                  mapper
+                                      .createObjectNode()
+                                      .set("c", mapper.createObjectNode().put("$eq", "c")))
+                              .add(
+                                  mapper
+                                      .createObjectNode()
+                                      .set("d", mapper.createObjectNode().put("$eq", "d"))))));
```

Comment:
Would it be more readable to parse this JSON doc from a string?

Reply:
Not a bad idea, let's have this in all test cases, would be more understandable I guess.. :+1:

Reply:
Done.